recipes/rmm/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.52.0"
+
+
+class RmmConan(ConanFile):
+    name = "rmm"
+    description = "RAPIDS Memory Manager"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rapidsai/rmm"
+    topics = ("cuda", "memory-management", "memory-allocation", "rapids", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        # Based partially on https://github.com/rapidsai/rmm/tree/v23.06.00#get-rmm-dependencies
+        return {
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "9.3",
+            "clang": "8",
+            "apple-clang": "14.0",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("thrust/1.17.2")
+        self.requires("spdlog/1.11.0")
+        self.requires("fmt/9.1.0")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if "backend" not in self.options["thrust"] or self.options["thrust"].backend != "cuda":
+            self.output.warning(
+                "RMM requires the CUDA backend to be enabled in Thrust by setting thrust/*:backend=cuda."
+            )
+
+    def _write_version_header(self):
+        # Workaround for the `rapids_cmake_write_version_file(include/rmm/version_config.hpp)` CMakeLists.txt step
+        # https://github.com/rapidsai/rapids-cmake/blob/branch-23.08/rapids-cmake/cmake/write_version_file.cmake
+        # https://github.com/rapidsai/rapids-cmake/blob/branch-23.08/rapids-cmake/cmake/template/version.hpp.in
+        major, minor, patch = self.version.split(".")[:3]
+        header_path = Path(self.source_folder) / "include" / "rmm" / "version_config.hpp"
+        header_path.parent.mkdir(parents=True, exist_ok=True)
+        header_path.write_text(
+            textwrap.dedent(
+                f"""\
+            #pragma once
+            #define RMM_VERSION_MAJOR {int(major)}
+            #define RMM_VERSION_MINOR {int(minor)}
+            #define RMM_VERSION_PATCH {int(patch)}
+            """
+            )
+        )
```

Comment:
```suggestion
        save(self, os.path.join(self.source_folder, "include", "rmm", "version_config.hpp")
            textwrap.dedent(
                f"""\
            #pragma once
            #define RMM_VERSION_MAJOR {Version(self.version).major}
            #define RMM_VERSION_MINOR {Version(self.version).minor}
            #define RMM_VERSION_PATCH {Version(self.version).patch}
            """
            )
        )
```

Conan tools offers [save](https://docs.conan.io/2/reference/tools/files/basic.html#conan.tools.files.files.save) that safer than using regular methods, because it also considers write permission in the folder (rare but we have problems in Windows).